### PR TITLE
packaging: replace `arch` with `uname -m`

### PR DESCRIPTION
**Pull request by neonsea**
All instances of the deprecated `arch` command are now replaced with `uname -m`.

Fixes: #423

Signed-off-by: Rasmus Moorats <me@neonsea.uk>

### CODE REVIEW

**Comment by devimc** [comment]
@neonsea thanks for the pull request , it works in x86_64, but not sure if it's the same in other platforms, like ppc, s390x and arm, cc @Pennyzct  @alicefr @Weichen81 @nitkon

**Comment by neonsea** [comment]
According to GNU docs, the two are equivalent. https://www.gnu.org/software/coreutils/manual/html_node/arch-invocation.html

**Comment by neonsea** [comment]
Unsure why the CI build failed, requires a version bump?

**Comment by devimc** [comment]
@neonsea https://github.com/kata-containers/packaging/blob/master/kernel/kata_config_version

**Comment by Pennyzct** [comment]
Hi~ @devimc thanks for the heads-up. It also works well on AArch64. ;)
```
root@entos-thunderx2-desktop:~/go/src/github.com/kata-containers/runtime# arch
aarch64
root@entos-thunderx2-desktop:~/go/src/github.com/kata-containers/runtime# uname -m
aarch64

```

**Comment by alicefr** [comment]
hi, for `s390x` should be fine except for the kernel path. Unfortunately, in linux kernel the path is `arch/s390`, but I'm not sure if the script `kernel/build-kernel` needs that. We can check (cc @tuan-hoang1 )

**Comment by jcvenegas** [comment]
hi @neonsea  this is looking good, could you bump the file `kata_config_version`.

**Comment by neonsea** [comment]
@jcvenegas sure, do you want it rebased into the one commit or is the version bump fine as a separate commit?

**Comment by jcvenegas** [comment]
A second commit is fine, just in case we need to bump again or revert (I hope this is not he case :) ).

**Comment by devimc** [comment]
@jcvenegas please take a look, @neonsea please bump the version again

**Comment by devimc** [comment]
@neonsea please squash the last commit

**Comment by neonsea** [comment]
Of course, sorry, I hope a reset/force push is OK.

**Comment by devimc** [comment]
@neonsea thanks, once CI is happy I'll merge your pr

File: kernel/kata_config_version [hunk 1]
```diff
@@ -1 +1 @@
-31
+33
```
Comment (jcvenegas) [inline kernel/kata_config_version]: I am sorry the file already has `33` better you rebase master and then bump to `34`.

**Comment by jodh-intel** [comment]
Related: https://github.com/kata-containers/osbuilder/pull/227.

Hi @neonsea - please could you rebase this PR to fix the version file conflict? We can then get this landed.

**Comment by neonsea** [comment]
Hi @jodh-intel! Sorry for the delay, have been too busy with uni stuff. I've "rebased" the PR, hopefully this is it.

**Pull request merged**